src/coloring/acyclic_coloring.jl

```diff
@@ -0,0 +1,209 @@
+            if color[w] != 0
+                forbiddenColors[color[w]] = v
+            end
+        end
+        #enforces the second condition of acyclic coloring
+        for w in outneighbors(g, v)
+            if color[w] != 0 #colored neighbor
+                for x in outneighbors(g, w)
+                    if color[x] != 0 #colored x
+                        if forbiddenColors[color[x]] != v
+                            prevent_cycle(v, w, x, g, color, forbiddenColors, firstVisitToTree, set)
+                        end
+                    end
+                end
+            end
+        end
+
+        color[v] = min_index(forbiddenColors, v)
+
+        #grow star for every edge connecting colored vertices v and w
+        for w in outneighbors(g, v)
+            if color[w] != 0
+                grow_star(v, w, g, firstNeighbor, set, color)
+            end
+        end
+
+        #merge the newly formed stars into existing trees if possible
+        for w in outneighbors(g, v)
+            if color[w] != 0
+                for x in outneighbors(g, w)
+                    if color[x] != 0 && x != v
+                        if color[x] == color[v]
+                            mergeTrees(v, w, x, g, set)
+                        end
+                    end
+                end
+            end
+        end 
+    end
+
+    return color
+end
+
+"""
+        prevent_cycle(v::Integer,
+                    w::Integer, 
+                    x::Integer,
+                    g::LightGraphs.AbstractGraph,
+                    color::AbstractVector{<:Integer}, 
+                    forbiddenColors::AbstractVector{<:Integer},
+                    firstVisitToTree::Array{Tuple{Integer, Integer}, 1},
+                    set::DisjointSets{LightGraphs.Edge})
+
+Subroutine to avoid generation of 2-colored cycle due to coloring of vertex v,
+which is adjacent to vertices w and x in graph g. Disjoint set is used to store
+the induced 2-colored subgraphs/trees where the id of set is a key edge of g
+"""
+function prevent_cycle(v::Integer,
+                        w::Integer, 
+                        x::Integer,
+                        g::LightGraphs.AbstractGraph,
+                        color::AbstractVector{<:Integer}, 
+                        forbiddenColors::AbstractVector{<:Integer},
+                        firstVisitToTree::AbstractArray{<: Tuple{Integer, Integer}, 1},
+                        set::DisjointSets{LightGraphs.Edge})
+    
+    edge = find_edge(g, w, x)
+    e = find_root(set, edge)
+    p, q = firstVisitToTree[edge_index(g, e)]
+    if p != v
+        firstVisitToTree[edge_index(g, e)] = (v, w)
+    elseif q != w
+        forbiddenColors[color[x]] = v
+    end
+end
+
+"""
+        min_index(forbiddenColors::AbstractVector{<:Integer}, v::Integer)
+
+Returns min{i > 0 such that forbiddenColors[i] != v}
+"""             
+function min_index(forbiddenColors::AbstractVector{<:Integer}, v::Integer)
```

Comment:
Could this be replaced by calls to `findfirst(!isequal(v), forbidden_colors)`:
```julia
julia> findfirst(!isequal(3), [5, 4, 6, 3, 1])
1

julia> findfirst(!isequal(3), [3, 5, 4, 6, 3, 1])
2
```

Reply:
Yeah, the output is going to be the same for both the functions. But since under the hood even `findfirst()` is also doing essentially the same task of iterating over all elements till the predicate is true for some `u`, do we have any other reasons for preferring one over the other?